client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoLegacyPreparedStatements.java

```diff
@@ -0,0 +1,120 @@
+import java.util.concurrent.ExecutorService;
+
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static java.lang.String.format;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static org.testng.Assert.assertEquals;
+
+public class TestTrinoLegacyPreparedStatements
+{
+    private static final String TEST_CATALOG = "test_catalog";
+    private TestingTrinoServer server;
+    private ExecutorService executorService;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        Logging.initialize();
+        server = TestingTrinoServer.create();
+        server.installPlugin(new TpchPlugin());
+        server.createCatalog(TEST_CATALOG, "tpch");
+        server.installPlugin(new BlackHolePlugin());
+        server.createCatalog("blackhole", "blackhole");
+        server.waitForNodeRefresh(java.time.Duration.ofSeconds(10));
+        setupTestTables();
+        executorService = newCachedThreadPool(daemonThreadsNamed(getClass().getSimpleName() + "-%s"));
+    }
+
+    private void setupTestTables()
+            throws SQLException
+    {
+        try (Connection connection = createConnection("blackhole", "blackhole");
+                Statement statement = connection.createStatement()) {
+            assertEquals(statement.executeUpdate("CREATE SCHEMA blackhole.blackhole"), 0);
+            assertEquals(statement.executeUpdate("CREATE TABLE test_table (x bigint, y varchar)"), 0);
+        }
+    }
+
+    private Connection createConnection(Boolean legacyPreparedStatements)
+            throws SQLException
+    {
+        String url = format("jdbc:trino://%s", server.getAddress());
+        Properties properties = new Properties();
+        properties.setProperty("user", "test");
+        properties.setProperty("LegacyPreparedStatements", String.valueOf(legacyPreparedStatements));
+        return DriverManager.getConnection(url, properties);
+    }
+
+    private Connection createConnection(String catalog, String schema)
+            throws SQLException
+    {
+        String url = format("jdbc:trino://%s/%s/%s", server.getAddress(), catalog, schema);
+        return DriverManager.getConnection(url, "test", null);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown()
+            throws Exception
+    {
+        executorService.shutdownNow();
+        executorService = null;
+        server.close();
+        server = null;
+    }
+
+    private void testSetting(Boolean legacyPreparedStatements)
+            throws Exception
+    {
+        try (Connection connection = createConnection(legacyPreparedStatements)) {
+            try (PreparedStatement ps = connection.prepareStatement("SELECT * FROM blackhole.blackhole.test_table WHERE x = ? AND y = ?")) {
+                ps.setInt(1, 42);
+                ps.setString(2, "test");
+                ps.executeQuery();
+            }
+        }
+    }
+
+    @Test
+    public void testTrue()
```

Comment:
test a default (without value) as well

Reply:
I thought about this, but don't the existing tests already do this?

Reply:
With the recent test change this issue is now moot